src/security/ZeroSensitiveMemory.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 1996-2021 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_SRC_SECURITY_ZEROSENSITIVEMEMORY_H
+#define SQUID_SRC_SECURITY_ZEROSENSITIVEMEMORY_H
+
+#if HAVE_LIMITS_H
+#include <limits.h>
+#endif
+
+#if HAVE_STRING_H
+#include <string.h>
+#endif
+
+#include <stdexcept>
+
+namespace Security {
+
+inline static void
+ZeroSensitiveMemory(void *dst, const size_t len)
+{
+    if (!dst)
```

Comment:
I suggest accepting and ignoring nil destinations with zero lengths. It can make some of the callers simpler, safer, and/or faster.

```suggestion
    if (!len)
        return; // tolerate zero-length nil dst

    if (!dst)
```